### ISSUE TITLE
Statistics about users?

### ISSUE THREAD

**Issue by josebailo** [fract4d/gnofract4d#163]
Exists some kind of statistics or metrics about people using the application? Which kind of users use the application? How many?

I think having this metrics could be useful to know how to focus the development effort. For example, if the main kind of users are mathematicians maybe we should focus on speed the calculations up. Or if the users are people using it just for fun we should focus on create a better interface and usability (provide an installer could be a first approach).

If you don't have them a first step could be try to collect them 😃

**Comment by jsbaboon** [comment]
Yes optimise speed and create better interface and make install easier and make all functions work as expected. Obviously. Regardless of metrics.

**Comment by YuiiraX** [comment]
I will start working on this as I had time to study the code for a bit.
I have several questions I want to clarify first:

- As this is a Linux/MacOS only software without an installer, would it be a problem to interrupt the user on program startup
- Where are we going to aggregate the data once we have collected it as need to inform the user on what we are collecting and were we are storing the data.

I would try to achieve the following things:

1. Let the user decide if they want the stats collection to be on every time they start the program
2. Ask the user what kind of user they are on a startup pop screen as well
3. Track button sequence within the program from start to end
4. Track session time

If there's no objection I will start working on it and please let me know the preferred method of collection. We might also need an update on the privacy policy as well.

**Comment by DX-MON** [comment]
I'm fine with this as long as I have a way to turn it off if I wish and it's made clear that such data collection is happening before the first packets leave my machine.

That will also cover the project for GDPR which requires informed consent.
